src/renderer/components/SettingsModal/definition.tsx

```diff
@@ -0,0 +1,449 @@
+// Copyright (c) 2016-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {FormattedMessage} from 'react-intl';
+import type {IntlShape} from 'react-intl';
+
+import {localeTranslations} from 'common/utils/constants';
+
+import type {SettingsDefinition} from 'types/settings';
+
+import CheckSetting from './components/CheckSetting';
+import DownloadSetting from './components/DownloadSetting';
+import NotificationSetting from './components/NotificationSetting';
+import RadioSetting from './components/RadioSetting';
+import SelectSetting from './components/SelectSetting';
+import ServerSetting from './components/ServerSetting';
+import SpellCheckerSetting from './components/SpellCheckerSetting';
+import UpdatesSetting from './components/UpdatesSetting';
+
+const getLanguages = async (func: () => Promise<string[]>) => {
+    return (await func()).filter((language) => localeTranslations[language]).
+        map((language) => ({label: localeTranslations[language], value: language})).
+        sort((a, b) => a.label.localeCompare(b.label));
+};
+
+const definition: (intl: IntlShape) => Promise<SettingsDefinition> = async (intl: IntlShape) => {
+    return {
+        general: {
+            title: (
+                <FormattedMessage
+                    id='renderer.components.settingsPage.general'
+                    defaultMessage='General'
+                />
+            ),
+            icon: 'settings-outline',
+            settings: [
+                {
+                    id: 'autoCheckForUpdates',
+                    component: UpdatesSetting,
+                    condition: (await window.desktop.getLocalConfiguration()).canUpgrade,
```

Comment:
Would these calls take long enough that it would be worth it to parallelize the various awaits here with a `Promise.all`?

Reply:
No, these calls should all be almost instantaneous.